aiutils/tftools/placeholder_management.py

```diff
@@ -34,165 +30,185 @@ class PlaceholderManager():
 
     def __init__(self):
         self._placeholders = dict()
-        self.issparse = dict()
-
-    def add_placeholder(self, name, dtype, shape=None, sparse=False):
+        self.issparse = set()
+        self.islist = set()
+
+    def add_placeholder(
+            self,
+            name,
+            dtype,
+            shape=None,
+            size=None,
```

Comment:
Not sure size is the right name.  I didn't realize it was keying to list placeholders.

Perhaps list_len=None?